homeassistant/components/switchbot_cloud/coordinator.py

```diff
@@ -0,0 +1,50 @@
+"""SwitchBot Cloud coordinator."""
+from logging import getLogger
+
+from async_timeout import timeout
+from switchbot_api import CannotConnect, Device, InvalidAuth, Remote, SwitchBotAPI
+
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryAuthFailed
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN, SCAN_INTERVAL
+
+_LOGGER = getLogger(__name__)
+
+
+class SwitchBotCoordinator(DataUpdateCoordinator):
+    """SwitchBot Cloud coordinator."""
+
+    _api: SwitchBotAPI
+    _device_id: str
+    _should_poll = False
+
+    def __init__(
+        self, hass: HomeAssistant, api: SwitchBotAPI, device: Device | Remote
+    ) -> None:
+        """Initialize SwitchBot Cloud."""
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=SCAN_INTERVAL,
+        )
+        self._api = api
+        self._device_id = device.device_id
+        self._should_poll = not isinstance(device, Remote)
+
+    async def _async_update_data(self):
+        """Fetch data from API endpoint."""
+        if not self._should_poll:
+            return None
+        try:
+            _LOGGER.debug("Refreshing %s", self._device_id)
+            async with timeout(10):
+                status = await self._api.get_status(self._device_id)
+                _LOGGER.debug("Refreshing %s with %s", self._device_id, status)
+                return status
+        except InvalidAuth as err:
+            raise ConfigEntryAuthFailed from err
```

Comment:
correct me if I am wrong, but doesn't this trigger a reauth flow? You don't have one as far as I can see

Reply:
@joostlek That is true, what should I do instead ?

Reply:
Hmm, I wasn't completely sure. I am not sure if you have to do ConfigEntryNotReady or not. I have to check other integrations on what they do

Reply:
@joostlek Yeah, thanks for catching it, it actually create tons of errors

Reply:
AcccuWeather and Github just raise UpdateFailed, will do the same